4 - AnimalShelter/Code/Animal.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+
+
+namespace AnimalShelter.Code
+{
+    public class Animal
+    {
+        public Guid UniqueAnimalId { get; private set; }
+        public AnimalType AnimalType { get; set; }
+        public Animal(AnimalType animalType)
+        {
+            AnimalType = animalType;
+            UniqueAnimalId = Guid.NewGuid();
```

Comment:
Good use of the guid like this. Always has a value, always unique, can't be set outside - perfect usage.